itou/companies/admin.py

```diff
@@ -260,6 +266,146 @@ def approvals_list(self, obj):
         valid_count = Approval.objects.is_assigned_to(obj.id).valid().count()
         return format_html('<a href="{}">Liste des {} Pass IAE (dont {} valides)</a>', url, count, valid_count)
 
+    def get_urls(self):
+        urls = super().get_urls()
+        return [
+            path(
+                "transfer/<int:from_company_pk>",
+                self.admin_site.admin_view(self.transfer_view),
+                name="transfer_company_data",
+            ),
+            path(
+                "transfer/<int:from_company_pk>/<int:to_company_pk>",
+                self.admin_site.admin_view(self.transfer_view),
+                name="transfer_company_data",
+            ),
+        ] + urls
+
+    def transfer_view(self, request, from_company_pk, to_company_pk=None):
+        if not self.has_change_permission(request):
+            raise PermissionDenied
+
+        from_company = get_object_or_404(models.Company.objects, pk=from_company_pk)
+        to_company = get_object_or_404(models.Company, pk=to_company_pk) if to_company_pk is not None else None
+
+        transfer_data = {}
+        for transfer_field in transfer.TransferField:
+            spec = transfer.TRANSFER_SPECS[transfer_field]
+            if model_field := spec.get("model_field"):
+                from_data = [getattr(from_company, model_field.name)]
+                transfer_data[transfer_field] = {
+                    "data": from_data,
+                }
+            else:
+                from_data = transfer.get_transfer_queryset(from_company, to_company, spec)
+                transfer_data[transfer_field] = {
+                    "data": from_data,
+                }
+
+        if not to_company:
+            form = SelectTargetCompanyForm(
+                from_company=from_company,
+                admin_site=self.admin_site,
+                data=request.POST or None,
+            )
+            if request.POST and form.is_valid():
+                return redirect(
+                    reverse(
+                        "admin:transfer_company_data",
+                        kwargs={
+                            "from_company_pk": from_company.pk,
+                            "to_company_pk": form.cleaned_data["to_company"].pk,
+                        },
+                    )
+                )
+        else:
+            fields_choices = []
+            for transfer_field in transfer.TransferField:
+                spec = transfer.TRANSFER_SPECS[transfer_field]
+                if model_field := spec.get("model_field"):
+                    if transfer_data[transfer_field]["data"] == [getattr(to_company, model_field.name)]:
+                        transfer_data[transfer_field]["data"] = None
+                    else:
+                        fields_choices.append((transfer_field.value, transfer_field.label))
+                elif from_data := transfer_data[transfer_field]["data"]:
+                    plural = "s" if len(from_data) > 1 else ""
```

Comment:
nit: Petit trick de lisibilité.
```suggestion
                    s = "s" if len(from_data) > 1 else ""
```